### ISSUE TITLE
Review Strategy for Adapters/Plugins

### ISSUE THREAD

**Issue by cruhl** [Stability-AI/StableStudio#3]

**Comment by hlky** [comment]
[Missing sampler from `StableDiffusionInput`](https://github.com/Stability-AI/StableStudio/blob/5bc209cbce2139c8560b5378a395fecc42991513/packages/stablestudio-plugin/src/Plugin.ts#L213C37-L234) + no UI element for sampler

**Comment by cruhl** [comment]
@hlky Created a stub issue for that, agreed.

https://github.com/Stability-AI/StableStudio/issues/12

**Comment by cruhl** [comment]
https://github.com/Stability-AI/StableStudio/pull/36#issuecomment-1559660704

Discussing where plugins should live ^

**Comment by cruhl** [comment]
This is relevant for https://github.com/Stability-AI/StableStudio/pull/51 too.

**Comment by cruhl** [comment]
@hlky @jtydhr88 @brycedrennan

You are the first folks to submit new plugins, thanks!

I wanted to see what you all thought about this comment ([reposted](https://github.com/Stability-AI/StableStudio/pull/36#issuecomment-1559660704))...

> @hlky We're considering moving out most plugins from this monorepo into separate repositories so you don't need our approval.
> 
> Beyond WebGPU + SDWebUI (which we have the capacity to support first-party), I'm afraid of @KAJdev and myself becoming a bottleneck.
> 
> We'd maintain a list of recommended plugins in this project linking out to efforts like this for maximum visibility.
> 
> How does that sound?
> 
> Open to suggestions/ideation here!

Basically we want to support third-party plugins and would love to know how you think they should operate in the long-term.

**Comment by cruhl** [comment]
In terms of first-party plugins, i.e. plugins which live in this monorepo, we're thinking [stable-diffusion-webui](https://github.com/Stability-AI/StableStudio/issues/25) and [WebGPU](https://github.com/Stability-AI/StableStudio/issues/4) make sense to have out-of-the-box support for.

In the medium-term, [we're trying to improve the DX](https://github.com/Stability-AI/StableStudio/pull/60) for plugins. (Please leave comments in that PR if you have feedback!)

What do you think about hosting https://github.com/Stability-AI/StableStudio/pull/51 and https://github.com/Stability-AI/StableStudio/pull/36 in your own repositories and then we can maintain some sort of "verified" list in StableStudio to make it really simple for folks to use?

**Comment by brycedrennan** [comment]
I agree that it doesn't scale to have all the plugins in the one repo. Creates unnecessary burden and liability on you guys.  

I think if you really wanted to build an ecosystem you'd:
 - create a clear contract for the plugins
 - allow (limited) changes to the ui. mostly custom settings. I didn't see how to handle upscaling, bigger images, controlnet, among other things.
 - support multiple concurrent plugins
 - build some plugin infra such that:
      -  a user could browse and add plugins right from the UI (like in pycharm)
      -  developers could publish to the plugin repository
 - dogfood the whole thing by moving your plugins to their own repos as well (which also provides useful examples)

I'm fine with whatever you guys want to do. I was able to just include StableStudio into my python library. The whole things works just running `pip install imaginairy && aimg server`

**Comment by jtydhr88** [comment]
hi guys, for webui plugin, I created this extension for webui to extend rest apis https://github.com/jtydhr88/sd-webui-StableStudio, do you want to move it into Stability organization or it is ok to keep it under mine?

**Comment by cruhl** [comment]
@jtydhr88 For now it's fine to keep in your repo, we'll make a longer-term call once everyone is cool with the overall strategy.

**Comment by jtydhr88** [comment]
@cruhl great, once https://github.com/Stability-AI/StableStudio/pull/24 merged, I will send a PR to webui extension index repo https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions to introduce this extension
